### PR TITLE
Formplayer frontend async restore

### PR DESCRIPTION
**Pull request by benrudolph**
@wpride @orangejenny this enables us to do async restores in formplayer. There's a fair amount going on here, so happy to sit down to review it with someone. I tried my best to keep commits logical, so could be easiest to review that way.

async_restore

fyi @proteusvacuum your baby in action

### CODE REVIEW

**Comment by dimagimon** [comment]
There are 19 errors:

* corehq/apps/cloudcare/static/cloudcare/js/formplayer/app.js, line 219 - Missing semicolon. (semi)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/app.js, line 239 - 'progressMessage' is already defined (no-redeclare)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/app.js, line 244 - Missing trailing comma. (comma-dangle)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/app.js, line 262 - Missing semicolon. (semi)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/apps/util/progress_view.js, line 15 - Missing semicolon. (semi)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/entities/menu.js, line 38 - 'request' is defined but never used (no-unused-vars)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/entities/menu.js, line 65 - 'username' is defined but never used (no-unused-vars)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/entities/menu.js, line 117 - Missing semicolon. (semi)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/menu_list_test.js, line 42 - Missing trailing comma. (comma-dangle)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/menu_list_test.js, line 56 - Missing semicolon. (semi)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/menu_list_test.js, line 60 - 'promise' is not defined. (no-undef)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/menu_list_test.js, line 62 - Missing semicolon. (semi)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/menu_list_test.js, line 65 - Missing semicolon. (semi)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/menu_list_test.js, line 75 - Missing semicolon. (semi)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/menu_list_test.js, line 76 - 'promise' is not defined. (no-undef)
Missing semicolon. (semi)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/menu_list_test.js, line 79 - Missing semicolon. (semi)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/menu_list_test.js, line 82 - Missing semicolon. (semi)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/menu_list_test.js, line 89 - Missing semicolon. (semi)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/menu_list_test.js, line 93 - 'promise' is not defined. (no-undef)
Missing semicolon. (semi)

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/menu_list_test.js [hunk 1]
```diff
@@ -25,4 +25,73 @@ describe('Render a case list', function () {
...
+            };
+
+        });
+
+        afterEach(function() {
+            server.restore();
+            clock.restore()
+        });
+
+        it('Should execute an async restore', function() {
```
Comment (benrudolph) [inline corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/menu_list_test.js]: this test is a bit on the longer side, but i think does a good job of giving a higher level overview of how this works

**Comment by dimagimon** [comment]
There are 19 errors:

* corehq/apps/cloudcare/static/cloudcare/js/formplayer/app.js, line 219 - Missing semicolon. (semi)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/app.js, line 239 - 'progressMessage' is already defined (no-redeclare)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/app.js, line 244 - Missing trailing comma. (comma-dangle)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/app.js, line 262 - Missing semicolon. (semi)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/apps/util/progress_view.js, line 15 - Missing semicolon. (semi)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/entities/menu.js, line 38 - 'request' is defined but never used (no-unused-vars)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/entities/menu.js, line 65 - 'username' is defined but never used (no-unused-vars)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/entities/menu.js, line 117 - Missing semicolon. (semi)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/menu_list_test.js, line 42 - Missing trailing comma. (comma-dangle)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/menu_list_test.js, line 56 - Missing semicolon. (semi)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/menu_list_test.js, line 60 - 'promise' is not defined. (no-undef)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/menu_list_test.js, line 62 - Missing semicolon. (semi)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/menu_list_test.js, line 65 - Missing semicolon. (semi)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/menu_list_test.js, line 75 - Missing semicolon. (semi)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/menu_list_test.js, line 76 - 'promise' is not defined. (no-undef)
Missing semicolon. (semi)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/menu_list_test.js, line 79 - Missing semicolon. (semi)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/menu_list_test.js, line 82 - Missing semicolon. (semi)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/menu_list_test.js, line 89 - Missing semicolon. (semi)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/menu_list_test.js, line 93 - 'promise' is not defined. (no-undef)
Missing semicolon. (semi)

**Comment by dimagimon** [comment]
There are 19 errors:

* corehq/apps/cloudcare/static/cloudcare/js/formplayer/app.js, line 219 - Missing semicolon. (semi)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/app.js, line 239 - 'progressMessage' is already defined (no-redeclare)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/app.js, line 244 - Missing trailing comma. (comma-dangle)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/app.js, line 262 - Missing semicolon. (semi)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/apps/util/progress_view.js, line 15 - Missing semicolon. (semi)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/entities/menu.js, line 38 - 'request' is defined but never used (no-unused-vars)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/entities/menu.js, line 65 - 'username' is defined but never used (no-unused-vars)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/entities/menu.js, line 117 - Missing semicolon. (semi)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/menu_list_test.js, line 42 - Missing trailing comma. (comma-dangle)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/menu_list_test.js, line 56 - Missing semicolon. (semi)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/menu_list_test.js, line 60 - 'promise' is not defined. (no-undef)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/menu_list_test.js, line 62 - Missing semicolon. (semi)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/menu_list_test.js, line 65 - Missing semicolon. (semi)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/menu_list_test.js, line 75 - Missing semicolon. (semi)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/menu_list_test.js, line 76 - 'promise' is not defined. (no-undef)
Missing semicolon. (semi)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/menu_list_test.js, line 79 - Missing semicolon. (semi)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/menu_list_test.js, line 82 - Missing semicolon. (semi)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/menu_list_test.js, line 89 - Missing semicolon. (semi)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/menu_list_test.js, line 93 - 'promise' is not defined. (no-undef)
Missing semicolon. (semi)

**Comment by orangejenny** [comment]
Looks reasonable. Appreciate that your commits are decently organized.

**Comment by proteusvacuum** [comment]
Might be a bit late to the game, but is there a reason you aren't showing the user the actual progress? 
I think @amstone326 did an amazing job of this on mobile, and part of the reason for building out this feature was to give users more information about what was going on. 

Also curious about the language you've used here, as it deviates a bit from how this is handled on mobile. It might be worth taking a look at https://github.com/dimagi/commcare-android/pull/1348 to check out how aliza did this on mobile so that we have a similar experience for users who use both platforms (and so our messaging is consistent). 

All in all though, I think this looks great!

**Comment by benrudolph** [comment]
no reason, the motivation for the change was a little different (users literally couldn't use formplayer if their restore was too big), so focused less on the progress bar aspect. can update though! thanks for the feedback

**Pull request merged**